examples/pgbouncer/README.md

```diff
@@ -0,0 +1,260 @@
+cfssl gencert -ca cert -ca-key key server_csr.json | cfssljson -bare server
+```
+
+## Save the certificates as secrets
+
+With all the necessary certificates generated, we will save them as secrets:
+
+``` shell
+# First the CA cert
+kubectl create secret tls <YOUR-CA-SECRET> --key="ca-key.pem" --cert="ca.pem"
+
+# Next the server cert
+kubectl create secret tls <YOUR-SERVER-CERT-SECRET> --key="server-key.pem" \
+  --cert="server.pem"
+```
+
+## Containerize PgBouncer
+
+Some users may prefer to containerize PgBouncer themselves. For this example,
+we will make use of an open source container, [edoburu/pgbouncer][edoburu]. One
+nice benefit of `edoburu/pgbouncer` is that it will generate all the PgBouncer
+configuration based on environment variables passed to the container.
+
+## Deploy a PgBouncer Deployment and Service
+
+With PgBouncer containerized, we will now create a deployment with PgBouncer and
+the proxy as a sidecar.
+
+First, we mount our CA certificate and server certificate and private key,
+renaming the certificate secrets to `cert.pem` and server private key to
+`key.pem`:
+
+> [`pgbouncer_deployment.yaml`](pgbouncer_deployment.yaml#L15-L29))
+
+``` yaml
+volumes:
+- name: cacert
+  secret:
+    secretName: <YOUR-CA-SECRET>
+    items:
+    - key: tls.crt
+      path: cert.pem
+- name: servercert
+  secret:
+    secretName: <YOUR-SERVER-CERT-SECRET>
+    items:
+    - key: tls.crt
+      path: cert.pem
+    - key: tls.key
+      path: key.pem
+```
+
+Next, we specify volume mounts in our PgBouncer container where the secrets will
+be stored:
+
+> [`pgbouncer_deployment.yaml`](pgbouncer_deployment.yaml#L31-L41))
+
+``` yaml
+- name: pgbouncer
+  image: <PG-BOUNCER-CONTAINER>
+  ports:
+  - containerPort: 8080
+  volumeMounts:
+  - name: cacert
+    mountPath: "/etc/ca"
+    readOnly: true
+  - name: servercert
+    mountPath: "/etc/server"
+    readOnly: true
+```
+
+Then we configure PgBouncer through environment variables. Note: we use 5431 for
+`DB_PORT` to leave 5432 available.
+
+> [`pgbouncer_deployment.yaml`](pgbouncer_deployment.yaml#L31-L41))
+
+``` yaml
+env:
+- name: DB_HOST
+  value: "127.0.0.1"
+- name: DB_USER
+  valueFrom:
+    secretKeyRef:
+      name: <YOUR-DB-SECRET>
+      key: username
+- name: DB_PASSWORD
+  valueFrom:
+    secretKeyRef:
+      name: <YOUR-DB-SECRET>
+      key: password
+- name: DB_NAME
+  valueFrom:
+    secretKeyRef:
+      name: <YOUR-DB-SECRET>
+      key: database
+- name: DB_PORT
+  value: "5431"
+- name: CLIENT_TLS_SSLMODE
+  value: "require"
+- name: CLIENT_TLS_CA_FILE
+  value: "/etc/ca/cert.pem"
+- name: CLIENT_TLS_KEY_FILE
+  value: "/etc/server/key.pem"
+- name: CLIENT_TLS_CERT_FILE
+  value: "/etc/server/cert.pem"
+```
+
+For the PgBouncer deployment, we add the proxy as a sidecar, starting it on port
+5431:
+
+> [`pgbouncer_deployment.yaml`](pgbouncer_deployment.yaml#L70-L76))
+
+``` yaml
+- name: cloud-sql-proxy
+  image: gcr.io/cloudsql-docker/gce-proxy:1.22.0
```

Comment:
Can we add a comment here encouraging users to update frequently

Reply:
Added a comment both here and in the corresponding YAML file.